java/core/src/test/org/apache/orc/impl/TestReaderImpl.java

```diff
@@ -527,4 +527,37 @@ public void testWithoutCompressionBlockSize() throws IOException {
       }
     }
   }
+
+  @Test
+  public void testSargSkipPickupGroupWithoutIndex() throws IOException {
+    Configuration conf = new Configuration();
+    Path[] paths = new Path[] {
+        new Path(workDir, "TestOrcFile.testSargSkipPickupGroupWithoutIndexCPlusPlus.orc"),
+        new Path(workDir, "TestOrcFile.testSargSkipPickupGroupWithoutIndexJava.orc"),
+    };
+    for (Path path: paths) {
+      FileSystem fs = path.getFileSystem(conf);
+      try (ReaderImpl reader = (ReaderImpl) OrcFile.createReader(path,
+          OrcFile.readerOptions(conf).filesystem(fs))) {
+
+        SearchArgument sarg = SearchArgumentFactory.newBuilder()
+            .startNot()
+            .lessThan("x", PredicateLeaf.Type.LONG, 100000L)
```

Comment:
The test column data is [0-1023]. I think we should use a no-intersection condition to verify that the pickup is skipped.

Reply:
Thanks @guiyanakuang , done.